pkg/cmd/server/origin/handlers.go

```diff
@@ -168,6 +176,98 @@ func namespacingFilter(handler http.Handler, contextMapper kapi.RequestContextMa
 	})
 }
 
+// versionSkewFilter adds a filter that may deny PUT an POST requests since they stomp entire resource objects from skewed
+// oc clients, since we know that those clients will strip unknown fields which can lead to unexpected outcomes
+func (c *MasterConfig) versionSkewFilter(binaryInfo version.Info, handler http.Handler) http.Handler {
+	skewedClientPolicy := c.Options.PolicyConfig.SkewedClientPolicy
+	if skewedClientPolicy == configapi.AllowAll {
+		return handler
+	}
+	seg := strings.SplitN(binaryInfo.GitVersion, "-", 2)
```

Comment:
This is done upstream to build the user agent string.  Change it there too?  It would be a lot easier to do if the type wasn't unversioned.

So, same logic in both spots, new field duplicating information, or replumb the logic through both sides discrete fields?